navigator/runtime-experimental/src/main/java/com/freeletics/mad/navigator/compose/internal/StoreViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package com.freeletics.mad.navigator.compose.internal
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.freeletics.mad.navigator.internal.NavigationExecutor
+import com.freeletics.mad.navigator.internal.NavigationExecutorStore
+
+internal class StoreViewModel(
+    internal val globalSavedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    private val stores = mutableMapOf<StackEntry.Id, NavigationExecutorStore>()
+    private val savedStateHandles = mutableMapOf<StackEntry.Id, SavedStateHandle>()
+
+    fun provideStore(id: StackEntry.Id): NavigationExecutor.Store {
+        var store = stores[id]
+        if (store == null) {
+            store = NavigationExecutorStore()
+            stores[id] = store
+        }
+        return store
```

Comment:
Imo, `stores.getOrPut { }` is more concise

Reply:
thanks, I often forget about many of the collection functions in the stdlib 👍 [9225227](https://github.com/freeletics/mad/pull/377/commits/922522757030a159dc610a6f6dbe6417b6f7544b)

Reply:
😁 There are about 200 of them